actions/local_database_MOD.js

```diff
@@ -0,0 +1,335 @@
+              updateContents('dbvaluelabel', 'Value:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;')
+              break
+            case 'size':
+            case 'count':
+            case 'fetcheverything':
+            case 'indexes':
+            case 'clear':
+            case 'defer':
+            case 'array':
+              updateVisibility('dbpathdiv', true)
+              updateVisibility('dbvaluediv', false)
+              updateContents('dbpathlabel', 'Table string. Example: <b><u><i>servers</i></u></b>:')
+              updateContents('dbvaluelabel', 'Value:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;')
+              break
+            case 'increment':
+            case 'decrement':
+              updateVisibility('dbpathdiv', true)
+              updateVisibility('dbvaluediv', false)
+              updateContents('dbpathlabel', 'Path string (split path by .) Example: <b><u><i>servers.1234567890</i></u></b>:')
+              updateContents('dbvaluelabel', 'Value:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;')
+              break
+            case 'randomkey':
+              updateVisibility('dbpathdiv', true)
+              updateVisibility('dbvaluediv', true)
+              updateContents('dbpathlabel', 'Table string. Example: <b><u><i>servers</i></u></b>:')
+              updateContents('dbvaluelabel', 'Number of random keys:')
+              break
+            default:
+              updateContents('dbpathlabel', 'Path string (split path by .) Example: <b><u><i>servers.1234567890</i></u></b>:')
+              updateContents('dbvaluelabel', 'Value:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;')
+              updateVisibility('dbpathdiv', true)
+              updateVisibility('dbvaluediv', false)
+              break
+          }
+          break
+      }
+    }
+
+    async function wait (id, time) {
+      await new Promise(resolve => setTimeout(resolve, time))
+      glob.onChangeOperation(document.getElementById(id))
+    }
+
+    glob.onChangeFormat(document.getElementById('dbformat'))
+
+    wait('dboperation', 50) // used b/c custom html isn't immediately available.
+  },
+
+  async action (cache) {
+    const functhis = this
+    const data = cache.actions[cache.index]
+    const dboperation = data.dboperation
+    const dbpath = this.evalMessage(data.dbpath, cache)
+    const splitpath = dbpath.split('.')
+    const dbvalue = this.eval(this.evalMessage(data.dbvalue, cache), cache)
+    const dbformat = data.dbformat
+    const db = this.getMods().require(dbformat) // safe require.
+    let output
+
+    if (!dbpath || !splitpath[0]) throw new Error('No DB path provided.')
+
+    function finished () { // Easy way to hand-wave away any synchronous or asynchronous callback function.
+      const varName = functhis.evalMessage(data.varName, cache)
+      const storage = parseInt(data.storage)
+      functhis.storeValue(output, storage, varName, cache)
+      functhis.callNextAction(cache)
+    }
```

Comment:
Why is this being called twice on the end of the switch/case's instead of just being called at the end? And if you're going to call it at then, it doesn't have to be a separate function

Reply:
That was required from a previous edit you had me change, so I suppose it does not need to be a separate function anymore.